lib/mongo_mapper/eager_includer.rb

```diff
@@ -50,18 +50,14 @@ def eager_include
 
   attr_reader :association_name
 
-  def instance_variable_name
-    "@eager_loaded_#{association_name}"
+  def get_association_proxy(record)
+    record.send(:get_proxy, @association)
   end
 
   def setup_association(record, value)
-    code = <<-RUBY
-      def #{association_name}
-        #{instance_variable_name}
-      end
-    RUBY
-    record.instance_eval(code, __FILE__, __LINE__)
-    record.instance_variable_set(instance_variable_name, value)
+    association_proxy = get_association_proxy(record)
+    Object.instance_method(:instance_variable_set).bind(association_proxy).call(:@target, value)
```

Comment:
Can't you:

```association_proxy.target = value```

See:

https://github.com/mongomapper/mongomapper/blob/master/lib/mongo_mapper/plugins/associations/proxy.rb#L14